docs/contributing/testing-guidelines/unit-testing.md

```diff
@@ -0,0 +1,168 @@
+# Unit testing
+
+Autoware uses the `ament_cmake` framework to build and run tests. The same
+framework is also used to analyze the test results.
```

Comment:
[nits] unnatural position of a new line character
Could you check other sentences as well? :bow: 

```suggestion
Autoware uses the `ament_cmake` framework to build and run tests. The same framework is also used to analyze the test results.
```

Or
```suggestion
Autoware uses the `ament_cmake` framework to build and run tests.
The same framework is also used to analyze the test results.
```

Reply:
I believe we should add a new line only when there is a full stop or a comma.

Rationale:
- It's more readable and easier to review.
- We can notice long sentences.
  - It helps us comply with https://developers.google.com/style/translation#write-short,-clear,-and-precise-sentences

@LalithVipulananthan @maxime-clem What do you think?

Reply:
Done in 2a3302a4942ca0478cf9d86ba7f1555937c39afa
There is now a newline after full stops and commas, except in list items and if the comma is for a very small sentence.

Reply:
@maxime-clem Thank you for updating!
And I'm sorry for my unclear explanation. :bow: 
You don't have to add a new line after every comma. I meant you can add a new line for long sentences.

![image](https://user-images.githubusercontent.com/31987104/172148820-3fe70496-21c2-4116-93c5-093585a9a298.png)

Reply:
Ah, I'm sorry I probably misunderstood. You added a new line because the latter of the sentence is long. :bow: